flexget/plugins/sites/newznab.py

```diff
@@ -1,125 +1,615 @@
+                    config['category_string'] = ','.join(str(c) for c in categories)
         return config
 
-    def fill_entries_for_url(self, url, task):
-        entries = []
-        log.verbose('Fetching %s' % url)
+    def _is_tv_entry(self, entry):
+        if any(entry.get(field) for field in self._series_identifier_fields):
+            return True
+        elif any(entry.get(field) for field in self._movie_identifier_fields):
+            return False
+        else:
+            return None
 
-        try:
-            r = task.requests.get(url)
-        except task.requests.RequestException as e:
-            log.error("Failed fetching '%s': %s" % (url, e))
+    def _has_meta_id(self, entry):
+        return any(entry.get(field) for field in self._metaid_fields)
+
+    def _has_supported_meta_id(self, entry):
+        if self._is_tv_entry(entry) is True:
+            return any(entry.get(field) for field in ['tvdb_id', 'tvrage_id'])
+        elif self._is_tv_entry(entry) is False:
+            return any(entry.get(field) for field in ['imdb_id'])
+        else:
+            return False
+
+    def update_metadata(self, entry, config):
+        if 'plugins_list' not in config:
+            return entry
+
+        show_lookup = self._is_tv_entry(entry)
+        if show_lookup is None:
+            log.warning('Could not determine entry type (tv/movie) for meta lookup: %s', entry)
+            return entry
 
-        rss = feedparser.parse(r.content)
-        log.debug("Raw RSS: %s" % rss)
+        search_list = []
+        if show_lookup is True:
+            if entry.get('series_name'):
+                title, year = split_title_year(entry['series_name'])
+                if year:
+                    search_list.insert(0, '%s (%s)' % (title, year))
+                list_append_unique(search_list, title, caseinsensitive=True)
+        elif show_lookup is False:
+            search_list.insert(0, entry['title'])
+            if entry.get('movie_name'):
+                title, year = split_title_year(entry['movie_name'])
+                if year:
+                    list_append_unique(search_list, '%s (%s)' % (title, year), caseinsensitive=True)
+                list_append_unique(search_list, title, caseinsensitive=True)
+                if entry.get('search_strings'):
+                    list_combine_unique(search_list, entry('search_strings'), caseinsensitive=True)
 
-        if not len(rss.entries):
-            log.info('No results returned')
+        search_entry = Entry(entry)
+        for search_string in search_list:
+            search_string = search_string.strip()
+            if not search_string:
+                continue
+            search_entry['title'] = search_string
+            # update entry metadata
+            if show_lookup is True:
+                search_entry['series_name'] = search_string
+                for plugin_name in config['plugins_list']:
+                    if plugin_name == 'trakt_lookup':
```

Comment:
I can shorten the ones that are identical, but "thetvdb_lookup" uses a extra parameter and trakt_lookup has "lazy_movie_lookup" instead of just "lookup".